sdk/storage/inc/blobs/block_blob_client.hpp

```diff
@@ -39,24 +39,30 @@ namespace Azure { namespace Storage { namespace Blobs {
         const std::string& blobUri,
         const BlockBlobClientOptions& options = BlockBlobClientOptions());
 
-    BlockBlobClient WithSnapshot(const std::string& snapshot);
+    BlockBlobClient WithSnapshot(const std::string& snapshot) const;
```

Comment:
`WithSnapshot` sounds like a member function that modifies the member instead of creating a new client. This is aligned with .NET but I still think .NET is not having a good name here.
`CloneWithSnapshot` maybe?

Reply:
I prefer `WithSnapshot` since it's aligned with .Net and Js